cmd/stack_create.go

```diff
@@ -100,87 +98,39 @@ The stack name must start with a lowercase letter, and can contain only lowercas
 			extractDir := filepath.Join(getHome(rootConfig), "extract")
 			extractDirFile := filepath.Join(extractDir, extractFilename)
 
-			// Get Repository directory and unmarshal
-			repoDir := getRepoDir(rootConfig)
-			var repoFile RepositoryFile
-			source, err := ioutil.ReadFile(filepath.Join(repoDir, "repository.yaml"))
+			stackEntry, err := getStackIndexYaml(repoID, stackID, rootConfig)
 			if err != nil {
-				return errors.Errorf("Error trying to read: %v", err)
-			}
-
-			err = yaml.Unmarshal(source, &repoFile)
-			if err != nil {
-				return errors.Errorf("Error parsing the repository.yaml file: %v", err)
+				return err
 			}
 
-			// get specificed repo and unmarshal
-			repoEntry := repoFile.GetRepo(repoID)
+			stackEntryURL := stackEntry.SourceURL
```

Comment:
you will need to verift that `stackEntry` is not nil, because in the case of dryrun the function `getStackIndexYaml` will return a nil pointer and this will fail. The tests gave it away